docs/developer-notes.md

```diff
@@ -0,0 +1,182 @@
+# Developer Notes
+
+## Copyright Header
+
+All files need to contain a copyright header. If you are using Android
+Studio, the copyright header is defined in the project settings and
+will usually be inserted automatically when creating a new file.
+
+Source code:
+
+```kotlin
+/*
+ * YacGuide Android application
+ *
+ * Copyright (C) <year> <Your Name>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+```
+
+XML (layout) file:
+```xml
+<!--
+ YacGuide Android application
+
+ Copyright (C) <year> <Your Name>
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ -->
+```
+
+
+## Development Script
+
+For automation of certain development steps, a Python based
+[script](../scripts/make) is available. The script uses the projects
+[Docker environment](#docker-environment).
+
+:exclamation: **Always start the script from the project root like
+`scripts/make <cmd>`.**
+
+
+### Tool Requirements
+
+The tool chain is tested and designed to work on Linux only. The
+following tools must be installed on the user machine to run this
+script:
+
+* Python 3 and the pip packages listed in file
+  [requirements.txt](../requirements.txt)
+* Docker client
+
+
+## Docker Environment
```

Comment:
I still don't get why this is necessary? In the `release-process.md` you don't mention docker at all but here you say, it's necessary for making releases.

Reply:
The user does not need to know this. This why a wrote all the Python code.